models/user.model.js

```diff
@@ -0,0 +1,48 @@
+const mongoose  = require('mongoose');
+const constants = require('../utils/constants');
+
+const userSchema = new mongoose.Schema({
+
+    name : {
+        type : String,
+        required : true
+    },
+    userId : {
+        type : String,
+        required : true,
+        unique : true
+    },
+    password : {
+        type : String,
+        required : true 
+    },
+    email : {
+        type : String,
+        required : true,
+        lowercase : true,
+        minLength : 10,
+        unique : true
+    },
+    createdAt : {
+        type : Date,
+        immutable :true,
+        default : () =>{
+            return Date.now()
+        }
+    },
+    updatedAt : {
+        type : Date,
+        default : () =>{
+            return Date.now()
+        }
+    },
+    userType : {
+        type : String,
+        required : true,
+        default : constants.userTypes.customer,
+        enum : [constants.userTypes.customer,constants.userTypes.admin,constants.userTypes.thearte_owner]
+    },
+    
```

Comment:
1 to M relation between user and theater he created ( only for the user of type theatre owner )